TLM/TMPE.API/Traffic/Enums/JunctionRestrictionRules.cs

```diff
@@ -0,0 +1,10 @@
+namespace TrafficManager.API.Traffic.Enums {
+    public enum JunctionRestrictionRules {
```

Comment:
This enum and its members need to be changed back to their original names, for two reasons:

1. The use of the word "Flags" was intentional so that it could be used for things that are related to junction restrictions but are not themselves "rules" or "restrictions."
2. The word "Allowed" needs to stay, because removing it makes the names semantically incorrect. Since they are members of a type named "restrictions," removing the word "Allowed" causes "restriction" to become the nearest contextual clue to their function. But a "restriction" suggests that setting it causes the action to *not* be allowed, which is the opposite of how the flags work.

Reply:
>  privately implemented 

why didn't I thought of that?!

Reply:
> > privately implemented
> 
> why didn't I thought of that?!

Dunno. ;-)

But are you sure implementing it privately is really necessary? If the patch is done right, I wouldn't expect it to care if there's an overload.

Reply:
Then again, if you're unsure, better safe than sorry.

Reply:
I am sure because I tested :)

Reply:
>  If the patch is done right, I wouldn't expect it to care if there's an overload.

Its not done right. it assumes there is no overload.

Reply:
GetMethod fails if there are overloads.

Reply:
BTW, you might have already known this, but in #1526, the way `NetManagerEvents` does patching accounts for overloading, by being very specific about the argument list.